### PR TITLE
add_more_support_for_print_source

### PR DESCRIPTION
**Pull request by xiaodwan**
Signed-off-by: Xiaodai Wang <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
In the cfg file, we only test esx, but the source code supports all sources(kvm, xen, esx).

**Comment by xiaodwan** [comment]
(1/3) type_specific.io-github-autotest-libvirt.v2v_options.positive_test.print_source.input_mode.libvirt.esx.esx_67.default.output_mode.none: PASS (12.97 s)
 (2/3) type_specific.io-github-autotest-libvirt.v2v_options.positive_test.print_source.input_mode.libvirt.xen.output_mode.none: PASS (13.32 s)
 (3/3) type_specific.io-github-autotest-libvirt.v2v_options.positive_test.print_source.input_mode.libvirt.kvm.default.output_mode.none: PASS (8.05 s)

**Comment by xiaodwan** [comment]
@chloerh @chunfuwen please help to review it. thanks!

**Comment by chloerh** [comment]
LGTM

**Comment by chloerh** [comment]
O, please modify this commit message, it's not beautiful

File: v2v/tests/src/v2v_options.py [hunk 1]
```diff
@@ -275,41 +276,85 @@ def check_source(output):
...
+            close_virsh = True
+
         # Check single values
         fail = []
-        xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+        try:
+            xml = vm_xml.VMXML.new_from_inactive_dumpxml(
+                vm_name, virsh_instance=v2v_virsh)
+        finally:
+            if close_virsh:
```
Comment (chunfuwen) [inline v2v/tests/src/v2v_options.py]: v2v_virsh could be None in some cases, better check it before call its method

File: v2v/tests/src/v2v_options.py [hunk 2]
```diff
@@ -275,41 +276,85 @@ def check_source(output):
...
                 fail.append(key)
 
         # Check disk info
         disk = list(xml.get_disk_all().values())[0]
-        bus, type = disk.find('target').get(
-            'bus'), disk.find('driver').get('type')
-        path = disk.find('source').get('file')
-        disks_info = "%s (%s) [%s]" % (path, type, bus)
+
+        def _get_disk_attr_value(obj, attr, subattr):
```
Comment (chunfuwen) [inline v2v/tests/src/v2v_options.py]: change _get_disk_attr_value to _get_disk_subelement_attr_value

File: v2v/tests/src/v2v_options.py (hunk 1 above)
Comment (xiaodwan) [reply v2v/tests/src/v2v_options.py]: The case will fail if virsh.VirshPersistent fail. so v2v_virsh could not be None. I didn't see any checks about this in other codes.

File: v2v/tests/src/v2v_options.py (hunk 2 above)
Comment (xiaodwan) [reply v2v/tests/src/v2v_options.py]: done

**Pull request merged**